server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java

```diff
@@ -53,8 +55,14 @@ public void getCheckpointMetadata(
         // TODO: Need to figure out a way to pass this information for segment metadata via remote store.
         final Version version = indexShard.getSegmentInfosSnapshot().get().getCommitLuceneVersion();
         try {
-            metadataMap = remoteDirectory.readLatestMetadataFile()
-                .getMetadata()
+            RemoteSegmentMetadata mdFile = remoteDirectory.readLatestMetadataFile();
+            // During initial recovery flow, the remote store might not have metadata as primary hasn't uploaded anything yet.
+            if (mdFile == null && indexShard.state().equals(IndexShardState.STARTED) == false) {
```

Comment:
what about relocation case? If md is null we would want to bail here or break relocation elsewhere?

Reply:
Added tracking issue for follow up: https://github.com/opensearch-project/OpenSearch/issues/8681

Reply:
I don't think my concern here makes sense, we would not be going through the peer recovery if the shard were started so we are ok.